consensus-types/blocks/rosidecar.go

```diff
@@ -0,0 +1,96 @@
+package blocks
+
+import (
+	"github.com/pkg/errors"
+)
+
+// ROSidecar represents a read-only sidecar with its block root.
+type ROSidecar struct {
```

Comment:
@kasey this it the new data structure I introduced to solve the "slice of interfaces" golang issue.